src/app/reducers/questions.ts

```diff
@@ -49,12 +53,18 @@ export function reducer(state = initialState, action: questions.Actions): State
       const points = calculateUserPoints(state.pickedQuestions[questionIndex], action.answers);
       const newPointsState = [...state.questionPoints];
       newPointsState[questionIndex] = points;
+      // all previous answers filtered out from new state and new answers added
+      const newAllAnswersState = state.allAnswers
+        .filter(existingAnswer => !action.answers.some(newAnswer => existingAnswer.optionId === newAnswer.optionId));
+      action.answers.forEach(replacingAnswer => newAllAnswersState.push(replacingAnswer));
+
       const newState = {
```

Comment:
This is a bit overcomplicated for a simple replace. Why not group the answers per question, for example `{ questionId: Answer[] }` or something? That way you can use something like `allAnswers[questionId] = replacingAnswers`